### PR TITLE
Matrix addition

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the new behavior (if this is a feature change)?**
We can add 2 matrices with the same dimension.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:
A new release (1.1.0) of math native is needed.

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: math/src/test/java/com/powsybl/math/matrix/AbstractMatrixTest.java [hunk 1]
```diff
@@ -102,6 +102,36 @@ public void testMultiplication() {
...
+        4 0
+        5 0
+        0 6
+         */
+        Matrix b = getMatrixFactory().create(3, 2, 3);
+        b.set(0, 0, 4);
+        b.set(1, 0, 5);
+        b.set(2, 1, 6);
+        Matrix cs = a.add(b);
+        DenseMatrix c = cs.toDense();
```
Comment (sylvlecl) [inline math/src/test/java/com/powsybl/math/matrix/AbstractMatrixTest.java:123]: Maybe we could check that in the sparse representation we still have one "missing" value, therefore a values array of size 5 and not 6.

**Review by sylvlecl — APPROVED** [review]

File: math/src/main/java/com/powsybl/math/matrix/Matrix.java [hunk 2]
```diff
@@ -226,13 +226,34 @@ default void addValue(int i, int j, double value) {
...
+    /**
+     * Addition the matrix with another one (alpha * this + beta * other). The resulting matrix has the same
+     * implementation as this matrix.
+     *
+     * @param other the other matrix
+     * @param alpha a scalar to multiply this matrix
+     * @param beta a scalar to multiply other matrix
+     * @return the result of the addition of this matrix and the other one
+     */
+    Matrix add(Matrix other, double alpha, double beta);
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply math/src/main/java/com/powsybl/math/matrix/Matrix.java:246]: The problem is that with elementary operations, you would have additional copies of matrices if you want to have such syntax, which would be problematic for large matrices.
```java
var n = m.times(alpha).plus(x.times(beta))
```

Or you just implement in place scalar multiply:
```java
m.times(alpha);
x.times(beta)
var n = m.plus(x)
```
But the api looks like less convenient?

File: math/src/test/java/com/powsybl/math/matrix/AbstractMatrixTest.java (hunk 1 above)
Comment (geofjamg) [reply math/src/test/java/com/powsybl/math/matrix/AbstractMatrixTest.java:123]: Done

File: math/src/main/java/com/powsybl/math/matrix/Matrix.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sylvlecl) [reply math/src/main/java/com/powsybl/math/matrix/Matrix.java:246]: Yes, thought about it but left the comment anyway. Maybe the "in place" behaviour could be an option to be used only when we want to optimize performances. Whatever, it's clearly not needed for now so no use to spend time on that :)

**Pull request merged**